aiosmtpd/smtp.py

```diff
@@ -584,9 +805,6 @@ async def smtp_RCPT(self, arg):
             await self.push(syntaxerr)
             return
         address, params = self._getaddr(arg)
-        if address is None:
-            await self.push(syntaxerr)
-            return
```

Comment:
`not address` already implictly includes `address is None`. Since there is no need to differ if address is None or bool, let's just merge the two into one. Should make `smtp_RCPT` a tad faster.